test/package/test_torchscript.py

```diff
@@ -0,0 +1,626 @@
+from unittest import skipIf
+from io import BytesIO
+from tempfile import TemporaryDirectory
+
+import torch
+from torch.package import PackageExporter, PackageImporter
+from torch.testing._internal.common_utils import run_tests, IS_FBCODE, IS_SANDCASTLE
+
+try:
+    from torchvision.models import resnet18
+
+    HAS_TORCHVISION = True
+except ImportError:
+    HAS_TORCHVISION = False
+skipIfNoTorchVision = skipIf(not HAS_TORCHVISION, "no torchvision")
+
+
+try:
+    from .common import PackageTestCase
+except ImportError:
+    # Support the case where we run this file directly.
+    from common import PackageTestCase
+
+from pathlib import Path
+
+packaging_directory = Path(__file__).parent
+
+
+class PackageScriptModuleTest(PackageTestCase):
+    """ScriptModule saving and loading in torch.Package tests."""
+
+    def test_save_scriptmodule(self):
+        """
+        Test basic saving of ScriptModule.
+        """
+        from package_a.test_module import ModWithTensor
+
+        scripted_mod = torch.jit.script(ModWithTensor(torch.rand(1, 2, 3)))
+
+        buffer = BytesIO()
+        with PackageExporter(buffer, verbose=False) as e:
+            e.save_pickle("res", "mod.pkl", scripted_mod)
+
+        buffer.seek(0)
+        importer = PackageImporter(buffer)
+        loaded_mod = importer.load_pickle("res", "mod.pkl", map_location="cpu")
+        input = torch.rand(1, 2, 3)
+        self.assertEqual(loaded_mod(input), scripted_mod(input))
+
+    @skipIf(
+        IS_FBCODE or IS_SANDCASTLE,
+        "Tests that use temporary files are disabled in fbcode",
+    )
+    def test_save_scriptmodule_file(self):
```

Comment:
This test is for testing to make sure TS works in a file instead of in just a buffer. The test below it needs a file also to create the directory out of (the rest of the tests all use buffers)

Reply:
ah I see, cool